webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/queries/DynGroupService.scala

```diff
@@ -112,18 +119,33 @@ class DynGroupServiceImpl(
     for {
       con         <- ldap
       entries     <- con.searchSub(rudderDit.GROUP.dn, dynGroupFilter, dynGroupAttrs:_*)
-      dyngroupIds <- ZIO.foreach(entries) { entry =>
+      dyngroups <- ZIO.foreach(entries) { entry =>
                        mapper.entry2NodeGroup(entry).toIO.chainError(s"Can not map entry to a node group: ${entry}")
                      }
     } yield {
       // The idea is to sort group to update groups with a query based on other groups content (objecttype group) at the end so their base group is already updated
       // This does not treat all cases (what happens when you have a group depending on a group which also depends on another group content)
       // We will sort by number of group queries we have in our group (the more group we depend on, the more we want to update it last)
       def numberOfQuery(group : NodeGroup) = group.query.map( _.criteria.filter(_.objectType.objectType == "group").size).getOrElse(0)
-      dyngroupIds.sortBy(numberOfQuery)
+      dyngroups.sortBy(numberOfQuery)
     }
   }.toBox
 
+  def getAllDynGroupsWithandWithoutDependencies(): Box[(Seq[NodeGroupId], Seq[NodeGroupId])] = {
+    (for {
+      con         <- ldap
+      entries     <- con.searchSub(rudderDit.GROUP.dn, dynGroupFilter, dynGroupAttrs:_*)
```

Comment:
here we should not take the nodeids, it would be even faster
it takes several seconds to fetch data when there is a large number of groupes

Reply:
ha, it is already not fetching it